docs/go-quickstart.md

```diff
@@ -0,0 +1,184 @@
+
+```cue
+package ship
+
+import "github.com/grafana/thema"
+
+thema.#Lineage
+name: "ship"
+seqs: [
+	{
+		schemas: [
+			// v0.0
+			{
+				// name is what we call the ship, and what's written in big letters on its hull
+				name: string
+				// masts is the number of masts the ship has. No fully rigged ship
+				// has ever had more than 7: https://oceannavigator.com/the-most-masted-schooner-ever-built/
+				masts: uint8 & < 8
+			},
+		]
+	},
+]
+```
+
+## Generating Go Types
+
+Once we have a schema defined, We can use it to generate a type in Go. Use the below command to generate a Go type
+
+
+```
+thema lineage gen gotypes -l ship.cue > go_type.go 
+```
+
+This should generate a go file has a type `ship` defined . The generated file looks like below
+
+```go
+// This file is autogenerated. DO NOT EDIT.
+//
+// Generated by "thema lineage gen" from lineage defined in ship.cue
+
+package ship
+
+// Ship defines model for ship.
+type Ship struct {
+	// masts is the number of masts the ship has. No fully rigged ship
+	// has ever had more than 7: https://oceannavigator.com/the-most-masted-schooner-ever-built/
+	Masts int `json:"masts"`
+
+	// name is what we call the ship, and what's written in big letters on its hull
+	Name string `json:"name"`
+}
+```
+
+
+CUE types are more expressive than Go types. To use the rich information from CUE in Go programs, users can use bindings. Go bindings provide access to the thema Lineage defined in `ship.cue`, and validate data against schemas in the lineage.
+
+## Generating Go Bindings
+
+Use the below command to generate a Go binding from `ship.cue`
+
+```
+thema lineage gen gobindings -l ship.cue > go_bindings.go
+```
+
+The generated Go bindings will look like this (generated comments have been removed):
+
+```go
+package ship
+
+import (
+	"embed"
+	"path"
+
+	"github.com/grafana/thema"
+	"github.com/grafana/thema/load"
+)
+
+//go:embed ship.cue
+var themaFS embed.FS
+
+func Lineage(rt *thema.Runtime, opts ...thema.BindOption) (thema.Lineage, error) {
+	// Load a build instance from the embedded fs
+	inst, err := load.InstancesWithThema(themaFS, path.Dir("ship.cue"))
+	if err != nil {
+		return nil, err
+	}
+
+	raw := rt.Context().BuildInstance(inst)
+	return thema.BindLineage(raw, rt)
+}
+
+// type guards
+
+var _ thema.LineageFactory = Lineage
+```
+We now have a single Go function our program can call, and it will load up the `ship.cue` from disk and return a [`thema.Lineage`](https://pkg.go.dev/github.com/grafana/thema#Lineage)
+We can then use the Lineage function to validate our data against the schema of choice
+
+## Validating data
+
+Using the Thema CLI, Users can validate the schema of choice defined in `ship.cue`. Use the below command to validate the schema against an invalid JSON data 
+
+```
+curl $URL > invalid.json
```

Comment:
What is $URL?

Reply:
Placeholder for now, but added the Url for once the PR is merged